### PR TITLE
create: add app templates doc and examples

### PR DESCRIPTION
**Pull request by psergee**

### CODE REVIEW

File: doc/examples.rst [hunk 1]
```diff
@@ -76,3 +76,80 @@ Or just one of them:
...
+    local login = {{.user_name}}
+
+    require("fiber").sleep(1)
+
+``MANIFEST.yaml``:
+
+.. code-block:: yaml
+
+    description: Simple app
+        vars:
```
Comment (0x501D) [inline doc/examples.rst:102]: Fix formatting indent levels

File: doc/examples.rst [hunk 2]
```diff
@@ -76,3 +76,80 @@ Or just one of them:
...
+            - prompt: User name
+            name: user_name
+            default: admin
+            re: ^\w+$
+
+Create a symlink in the template directory:
+``{{.name}} -> init.lua``
```
Comment (0x501D) [inline doc/examples.rst:109]: What does it mean? this is example, describe clearly.
e.g. Create a symlink in the template directory:
`` sh
ln -s target link_name
``

File: doc/examples.rst [hunk 3]
```diff
@@ -76,3 +76,80 @@ Or just one of them:
...
+
+Here is how the directory structure looks like::
+
+    ./
+    ├── tarantool.yaml
+    └── templates
+        └── simple
+            ├── init.lua.tt.template
+            ├── MANIFEST.yaml
+            └── {{.name}}.lua -> ./init.lua
```
Comment (0x501D) [inline doc/examples.rst:128]: it is not clear what the init.lua file contains, where is the description?

**Review by 0x501D — CHANGES_REQUESTED** [review]
The example is not clear, I could not finish the setup on it

File: doc/examples.rst (hunk 2 above)
Comment (psergee) [reply doc/examples.rst]: Removed

File: doc/examples.rst (hunk 3 above)
Comment (psergee) [reply doc/examples.rst]: Init.lua is just a sample of app code.

**Review by 0x501D — APPROVED** [review]
LGTM

**Review by LeonidVas — APPROVED** [review]
LGTM

**Pull request merged**